spec/controllers/users/phone_confirmation_controller_spec.rb

```diff
@@ -89,6 +92,18 @@
         it 'displays success flash notice' do
           expect(flash[:success]).to eq t('notices.phone_confirmation_successful')
         end
+
+        it 'tracks the update and confirmation event' do
+          expect(@analytics).to have_received(:track_event).
+            with('User confirmed their phone number')
+          expect(@analytics).to have_received(:track_event).
+            with('User changed and confirmed their phone number')
+        end
+
+        it 'creates a new :phone_changed event' do
+          expect(subject).to have_received(:create_user_event).with(:phone_changed)
+          expect(subject).to have_received(:create_user_event).exactly(:once)
+        end
```

Comment:
It does use the DB. Every single `it` block here creates a user in the DB via `sign_in_as_user` on line 65 in the `before` block at the top of the `describe #confirm` section. Testing analytics shouldn't require creating a user, but that's a refactor for another PR.